### ISSUE TITLE
Error: Hosting target (project-name) not detected in firebase.json

### ISSUE THREAD

**Issue by pflevy** [firebase/firebase-tools#2731]
### [REQUIRED] Environment info

**firebase-tools:**v8.13.1

**Platform:**macOS

### [REQUIRED] Test case

`firebase deploy --only hosting:(project-name)`

### [REQUIRED] Steps to reproduce

I'm using the following Github action: https://github.com/w9jds/firebase-action/ which uses the latest release of firebase-tools.
Since the last update, it's not working anymore. I had to manually set to an older version of firebase-tools (I went for v8.12.1, I haven't tried with v8.13.0)

### [REQUIRED] Expected behavior

Successful deploy

### [REQUIRED] Actual behavior
Deploy error
Error: Hosting target (project-name) not detected in firebase.json

**Comment by bkendall** [comment]
I am able to replicate this with `firebase deploy --only hosting:<project>`. Taking a look...

**Comment by bkendall** [comment]
Question for you @pflevy -

on `v8.12.1`, I actually cannot directly do `firebase deploy --only hosting:<project-name>`. The `--only hosting:<target>` syntax is actually for deploying to Hosting _targets_ in a multi-site setup. More information on that: https://firebase.google.com/docs/hosting/multisites#cli-commands-with-deploy-targets

Do you have those targets set up and are you using the correct name? I can poke around a bit more, but specifically doing `--only hosting:<SITE-name>` (which may be the same as the project name) is not supported.

**Comment by ludufre** [comment]
Same issue here with 8.13~

I use `firebase deploy --only hosting:sonar-tim-web` for almost 2 years.

```bash
ludufre@iMac sonar % npm i -g firebase-tools@8.13

npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
/Users/ludufre/.npm-packages/bin/firebase -> /Users/ludufre/.npm-packages/lib/node_modules/firebase-tools/lib/bin/firebase.js
+ firebase-tools@8.13.1
updated 1 package in 26.881s
ludufre@iMac sonar % firebase deploy --only hosting:sonar-tim-web

=== Deploying to 'sonar-tim'...

i  deploying hosting

Error: Hosting target sonar-tim-web not detected in firebase.json
ludufre@iMac sonar % npm i -g firebase-tools@8.12                
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
/Users/ludufre/.npm-packages/bin/firebase -> /Users/ludufre/.npm-packages/lib/node_modules/firebase-tools/lib/bin/firebase.js
+ firebase-tools@8.12.1
updated 1 package in 7.532s
ludufre@iMac sonar % firebase deploy --only hosting:sonar-tim-web

=== Deploying to 'sonar-tim'...

i  deploying hosting
i  hosting[sonar-tim-web]: beginning deploy...
i  hosting[sonar-tim-web]: found 311 files in platforms/browser/www
✔  hosting[sonar-tim-web]: file upload complete
i  hosting[sonar-tim-web]: finalizing version...
✔  hosting[sonar-tim-web]: version finalized
i  hosting[sonar-tim-web]: releasing new version...
✔  hosting[sonar-tim-web]: release complete

✔  Deploy complete!

Project Console: https://console.firebase.google.com/project/sonar-tim/overview
Hosting URL: https://sonar-tim-web.web.app
ludufre@iMac sonar % 
```

**Comment by ludufre** [comment]
Same issue here with 8.13~

I use `firebase deploy --only hosting:x-x-x` for almost 2 years.

```bash
ludufre@iMac ludufre % npm i -g firebase-tools@8.13

npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
/Users/ludufre/.npm-packages/bin/firebase -> /Users/ludufre/.npm-packages/lib/node_modules/firebase-tools/lib/bin/firebase.js
+ firebase-tools@8.13.1
updated 1 package in 26.881s
ludufre@iMac ludufre % firebase deploy --only hosting:x-x-x

=== Deploying to 'x-x'...

i  deploying hosting

Error: Hosting target x-x-x not detected in firebase.json
ludufre@iMac ludufre % npm i -g firebase-tools@8.12                
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
/Users/ludufre/.npm-packages/bin/firebase -> /Users/ludufre/.npm-packages/lib/node_modules/firebase-tools/lib/bin/firebase.js
+ firebase-tools@8.12.1
updated 1 package in 7.532s
ludufre@iMac ludufre % firebase deploy --only hosting:x-x-web

=== Deploying to 'x-x'...

i  deploying hosting
i  hosting[x-x-x]: beginning deploy...
i  hosting[x-x-x]: found 311 files in platforms/browser/www
✔  hosting[x-x-x]: file upload complete
i  hosting[x-x-x]: finalizing version...
✔  hosting[x-x-x]: version finalized
i  hosting[x-x-x]: releasing new version...
✔  hosting[x-x-x]: release complete

✔  Deploy complete!

Project Console: https://console.firebase.google.com/project/x-x/overview
Hosting URL: https://x-x-x.x.x
ludufre@iMac ludufre % 
```

**Comment by spenceclark** [comment]
I've had this same problem start happening from last night.
I use the following in my CI pipeline:

```
firerbase use PROJECTNAME --token $FIREBASE_TOKEN
firebase deploy --only hosting:SITE
```

I now get the error:

> Hosting target **SITE** not detected in firebase.json

Nothing else has changed on this project or CI defintion for months

**Comment by vin-fandemand** [comment]
Same here... since two days

**Comment by AnaWhibley** [comment]
Same here. I've always used `firebase deploy --only hosting:<name>`

**Comment by pflevy** [comment]
> Question for you @pflevy -
> 
> on `v8.12.1`, I actually cannot directly do `firebase deploy --only hosting:<project-name>`. The `--only hosting:<target>` syntax is actually for deploying to Hosting _targets_ in a multi-site setup. More information on that: https://firebase.google.com/docs/hosting/multisites#cli-commands-with-deploy-targets
> 
> Do you have those targets set up and are you using the correct name? I can poke around a bit more, but specifically doing `--only hosting:<SITE-name>` (which may be the same as the project name) is not supported.

In this specific case, I only use **one hosting** and I don't have the target set up on firebase.json. I actually tried adding it as
target: <project-name> but it didn't work.

Also, I do have another project where I rely on the same action and make use of **two hosts**. But I also don't have the 'target' set on firebase.json and only pass it as a parameter `deploy --only hosting:<SITE-name>`, which has always worked fine.

**Comment by stanleygomes** [comment]
Worked to me:

**Comment by stanleygomes** [comment]
Worked for me, this way:

```
npm install -g firebase-tools@8.12
```

Then:

```
firebase deploy -f --project my-project-name --only hosting:my-project-name
```

**Comment by bkendall** [comment]
Ack. Thanks everyone for the insights. If we regressed this behavior, makes sense to me that we should restore it. I'll look into it this morning.

**Comment by Angel-Q87** [comment]
I was presented with the same problem and using an older version of firebase-tools solved the problem. Thanks for your comments. Greetings.

**Comment by MichaelDCooper** [comment]
@bkendall How should hosting be specified until a fix is up?

**Comment by comeacross1989** [comment]
Having same issue on local windows machine. I was on an older version of firebase-tools (probably 7... something) then upgraded and that didn't fix the issue. Then installed version 8.12, added the "site" property to  my firebase.json file and then deployed as usual with "firebase deploy --only hosting:SITENAME"

**Comment by ThaDeveloper** [comment]
`npm install -g firebase-tools@8.12`
Then:
`firebase deploy --only hosting:project-name`
works for me

**Comment by matheuscouto** [comment]
```
npm install -g firebase-tools@8.12
firebase deploy --only hosting:project-name
```

works for me as well

**Comment by usamedin** [comment]
I get the same issue with no changes to firebase config

**Comment by mark-gene** [comment]
I also get the same issue with the latest version of firebase when installed using `curl -sL https://firebase.tools | bash`

using firebase-tools@8.12 worked for me as well.

**Comment by bkendall** [comment]
This will go out in the next release of firebase tools. Thanks for your patience.

**Comment by Teepath** [comment]
Having the same issue...

**Comment by alkeicam** [comment]
Same problem since 3 days using : 
firebase -P PROJ_NAME deploy -m "MESSAGE" --only hosting:SITE_NAME
results in

Error: Hosting target SITE_NAME not detected in firebase.json

**Comment by alkeicam** [comment]
> @bkendall How should hosting be specified until a fix is up?

Follow https://firebase.google.com/docs/hosting/multisites#set_up_deploy_targets

TL;DR
1. add target id to .firebaserc
2. add target entry to your firebase.json
3. deploy

it solved the problem for me with multiple sites in one hosting project

**Comment by bkendall** [comment]
This should be fixed in `v8.14.0`. Thanks for y'alls patience.

**Comment by RoCampos** [comment]
I had the same problem: **Hosting target (project-name) not detected in firebase.json**

I looked at .firabaserc and get the target name 

`targets": {
    "XXXXX": {
      "hosting": {
        "Teste": [
          "XXXXX"
        ]
      }
    }`

Then, I ran `firebase deploy --only hosting:XXXXX` and it worked.

**Comment by FelipeLucke** [comment]
The same problem here...

Now Node 12 have suport to firebase functions, but when I try to deploy my aplication this error hapen. My project in firebase have 2 sites in hosting and want to send my server to take 1 of then..

**Comment by sonicwim** [comment]
I tried the Angular tutorial from [https://angular.io/start](https://angular.io/start) and then also tried to deploy it from my local copy to Firebase but I also needed to specify the target as @RoCampos said.

And I'm using version 8.18.0, any idea why this is still happening?

**Comment by bkendall** [comment]
@sonicwim if you could open a new issue with more details, that would be super helpful. Please include all the relevant parts of `firebase.json` and `--debug` output.

**Comment by sonicwim** [comment]
I will do that.

**Comment by felansu** [comment]
happy new year

**Comment by sonicwim** [comment]
As I'm a noob on Github I could not find back this thread after posting my message so I could also not let you know that my problem is solved, it was a because I misinterpreted an instruction (about the default project name) off the tutorial.

Tips on navigating to your posts would be appreciated... ;-)

And I only found this page back because I got a mail today of someone saying happy new year but this seems to be removed already.

**Comment by usamedin** [comment]
> And I only found this page back because I got a mail today of someone saying happy new year but this seems to be removed already.

Yeah the Happy new year message confused me a lot

**Comment by sonicwim** [comment]
Is there a way, @usamedin, to see your forum activity? I can't seem to find it...

**Comment by usamedin** [comment]
@sonicwim try https://github.com/notifications/subscriptions, when you comment on a thread you are auto-subscribed to it

**Comment by maurocolella** [comment]
Has this ever been fixed?

**Comment by lericslh** [comment]
only open the directory and open firebase.json

and add "site": "<site-name>"

below the second line "hosting"

**Comment by dnishiyama** [comment]
I ended up here because I hadn't followed the full procedure for setting up multi-site hosting. This url had all the info I needed: https://firebase.google.com/docs/hosting/multisites

I needed to run `firebase hosting:sites:create SITE_ID`
Then `firebase target:apply hosting TARGET_NAME SITE_ID`
Then I had to modify firebase.json for multi-site hosting as suggested in Step 4: ```
"hosting": [ {
      "target": {TARGET_NAME},  // "blog" is the applied TARGET_NAME for the Hosting site "myapp-blog"
      "public": "blog/dist",
```
Hope that helps someone

**Comment by AndyKgotatso** [comment]
Found it!

Follow steps here - https://firebase.google.com/docs/cli/targets

1. Make sure you don't remove `"etags": {}` from the `.firebaserc` file

**Comment by Nano1408** [comment]
Ami me funciono cerrando sesión con el comando "firebase logout" y volver a iniciar sesión con "firebase login", esto refresca los CLI de recopilación de información, después le hice el despliegue normal sin generar el error "**Error:** Cannot deploy a web framework from source because the experiment webframeworks is not enabled. To enable webframeworks run firebase experiments:enable webframeworks".

Comparto pasos para los que le funcione:
Paso 1: Cerrar sesión "firebase logout"
paso 2: iniciar sesión "firebase login" -> esto refresca los CLI y Emulator Suite para los permisos en tu cuenta firebase
paso 3: desplegar cambios "firebase deploy"

Antes
<img width="1013" alt="Captura de Pantalla 2024-04-28 a las 1 20 12 a m" src="https://github.com/firebase/firebase-tools/assets/104662887/d0dd0ef4-d1ce-422d-a295-445e6d42d780">


Después
<img width="781" alt="Captura de Pantalla 2024-04-28 a las 1 20 55 a m" src="https://github.com/firebase/firebase-tools/assets/104662887/fb5076a9-5eb9-45fe-a343-82082b4e16bc">

**Comment by sarkissian001** [comment]
Try the following version `w9jds/firebase-action:v13.8.3`
